packages/cadl-ranch-specs/http/content-negotiation/main.tsp

```diff
@@ -0,0 +1,63 @@
+import "@typespec/http";
+import "@azure-tools/cadl-ranch-expect";
+import "@azure-tools/typespec-client-generator-core";
+
+using TypeSpec.Http;
+using Azure.ClientGenerator.Core;
+
+@doc("Test describing optionality of the request body.")
+@supportedBy("dpg")
+@scenarioService("/content-negotiation")
+namespace ContentNegotiation;
+
+@scenario
+@scenarioDoc("""
+  Scenario that returns a different file encoding depending on the accept header.
+
+  - image/png return a png image
+  - image/jpeg return a jpeg image
+  """)
+@route("same-body")
+namespace SameBody {
+  model PngImage {
+    @header contentType: "image/png";
+    @body image: bytes;
+  }
+
+  model JpegImage {
+    @header contentType: "image/jpeg";
+    @body image: bytes;
+  }
+
+  @sharedRoute
+  op getAvatarAsPng(@header accept: "image/png"): PngImage;
+
+  @sharedRoute
+  op getAvatarAsJpeg(@header accept: "image/jpeg"): JpegImage;
+}
+
+@scenario
+@scenarioDoc("""
+  Scenario that a different payload depending on the accept header.
+
+  - application/json return a png image in a Json object
+  - image/png return the png image
+  """)
+@route("different-body")
+namespace DifferentBody {
+  model PngImage {
+    @header contentType: "image/png";
+    @body image: bytes;
+  }
+
+  model PngImageAsJson {
+    @header contentType: "image/jpeg";
+    content: bytes;
+  }
+
+  @sharedRoute
+  op getAvatarAsPng(@header accept: "image/png"): PngImage;
+
+  @sharedRoute
+  op getAvatarAsJson(@header accept: "application/json"): PngImageAsJson;
```

Comment:
SDK would either require a "client.tsp", or these `getAvatarAs##` in different interface having different operation name.

As SDK would ignore interface by default, then these API would have conflict API name.

Reply:
Seems `SameBody` and `DifferentBody` are the same. It only return differently according to the accept header. I suppose those tow interface will set different input with different content-type.

Reply:
This testcase is only for multiple-content-type for response.  Shall we add tests for multiple content-type for input body parameter?

Reply:
yeah missed the `@operationGroup` on the namespaces, will add it.

Reply:
@chunyu3 that is a good point but I think it is a different scenario. Different input doesn't require negotiation for the output, it could just be represented as a union of input.

However I'll maybe try to rename this to make sure to leave room for the input scenarios